Docs 5947 fix syntax (#373)
mulesoft/docs-connectors: modules/ROOT/pages/sftp/sftp-list.adoc

include::_attributes.adoc[]
endif::[]

> The `list` operation returns a list of messages, where each message represents any file or folder found within the directory path (`directoryPath`).

Reviewer: ```suggestion
The `list` operation returns a list of messages in which each message represents any file or folder found within the directory path (`directoryPath`).
```

> By default, the operation does not read or list files or folders within any sub-folders of `directoryPath`.

Reviewer: ```suggestion
By default, the operation does not read or list files or folders within any subfolders of `directoryPath`.
```

> To list files or folders within any sub-folders, you can set the `recursive` parameter to `true`.

Reviewer: ```suggestion
To list files or folders within subfolders, set the `recursive` parameter to `true`.
```

> Set the path in the file configuration (through `<sftp:config/>`). The path can be absolute, or it can be relative to a working directory (`workingDir`). 

Reviewer: ```suggestion
Set the path in the file configuration by using `<sftp:config/>`. The path can be absolute, or it can be relative to a working directory (`workingDir`). 
```

> The following example lists the contents of a folder and all messages in a `directoryPath`, without listing the contents of any of the sub-folders. The `<foreach>` and `<choice>` components handle each directory in the list differently from the way it handles each file.

Reviewer: ```suggestion
The following example lists the contents of a folder and all messages in `directoryPath`, without listing the contents of the subfolders. The `<foreach>` and `<choice>` components handle each directory in the list differently from the way they handle each file.
```
Also, I wonder if "process" or "manage" is better than "handle"?

</flow>
----

== Poll a Directory

> Poll a directory to look for new files to process. Because Mule 4 no longer has a polling message source (like the Mule 3 transport did), you can poll in Mule 4 by combining a `<scheduler />` element with the `<sftp:list>` element.

Reviewer: ```suggestion
Poll a directory to look for new files to process. Because Mule 4 no longer has the polling message source that Mule 3 did, you poll in Mule 4 by combining a `<scheduler />` element with the `<sftp:list>` element.
```

</flow>
----

> In the above example, a flow lists the contents of a folder once per second. The flow then processes the files one by one and deletes each file after it is processed.

Reviewer: ```suggestion
In the previous example, a flow lists the contents of a folder once per second. The flow then processes the files one by one, deleting each file after it is processed.
```
I actually think that "this example" is fine if context makes clear that it is the one immediately preceding and not one immediately following (which would be introduced by text ending with a colon). Your choice.

> In Mule 4 (unlike in Mule 3), you have to manually delete or move files when the flow ends, which gives you a much higher level of control, especially in failure scenarios. You can also do more things such as moving, renaming, sending to an external system, and other actions, as well as change the behavior, depending on other conditions or failures.

Reviewer: ```suggestion
In Mule 4 you must manually delete or move files when the flow ends, giving you better control than you had in Mule 3, especially in failure scenarios. You can also do more things, such as moving, renaming, and sending to an external system, and you can change the behavior, depending on other conditions or failures.
```
But consider clarifying  what is moved, renamed, or sent to an external system (files?) and whats behavior you can change (the flow?). Also, to what are "other conditions" other?

== Match Filter

> When listing files, use the `<sftp:matcher>` element so that only files that match a certain criteria are considered. This element defines the possible criteria to use to either accept or reject a file.

Reviewer: ```suggestion
When listing files, use the `<sftp:matcher>` element so that only files that match the specified  `matcher` criteria are accepted.
```

maxSize="1024" />
----

=== Individual Attributes

> Individual attributes are optional and are ignored if not provided.  You can use an `AND` operator to combine individual attributes. 

Reviewer: ```suggestion
Individual attributes are optional and are ignored if you do not provide values for them. Use an `AND` operator to combine them. 
```

* `accessedSince` +

> Inclusive lower boundary for the file access stamp expressed as either a DateTime instance or a String in ISO-8601 format.

Reviewer: ```suggestion
An inclusive lower boundary for the file access stamp expressed as either a DateTime instance or a string in ISO-8601 format.
```

* `accessedUntil` + 

> Inclusive upper boundary for the file access stamp expressed as either a DateTime instance or a String in ISO-8601 format.

Reviewer: ```suggestion
An inclusive upper boundary for the file access stamp expressed as either a DateTime instance or a string in ISO-8601 format.
```

* `createdSince` +

> Inclusive lower boundary for the file creation stamp expressed as either a DateTime instance or a String in ISO-8601 format.

Reviewer: ```suggestion
An inclusive lower boundary for the file creation stamp expressed as either a DateTime instance or a string in ISO-8601 format.
```

* `createdUntil` +

> Inclusive upper boundary for the file creation stamp expressed as either a DateTime instance or a String in ISO-8601 format.

Reviewer: ```suggestion
An inclusive upper boundary for the file creation stamp expressed as either a DateTime instance or a string in ISO-8601 format.
```

* `directory` +
Match only if the file is a directory.
* `filenamePattern` +

> Similar to the current filename pattern filter but more powerful. Glob expressions (default) and regex are supported. You can select which one to use by setting a prefix. For example: `glob:*.{java, js}` or `regex:[0-9]test.csv`. 

Reviewer: ```suggestion
Similar to the current filename pattern filter but more powerful. Glob expressions (default) and regex are supported. You can select which one to use by setting a prefix: for example, `glob:*.{java, js}` or `regex:[0-9]test.csv`. 
```

* `maxSize` +
Inclusive upper boundary for the file size expressed in bytes.
* `minSize` +
Inclusive lower boundary for the file size expressed in bytes.
* `path-pattern` + 

> Same as filenamePattern but applies over the entire file path. Not just a filename.

Reviewer: Doublechecking the second sentence: are we saying that filenamePattern is not just a filename or that filenamePattern applies over an entire file path and not just a file?

* `regularFile` +
Match only if the file is a regular file.
* `symbolicLink` +
Match only if the file is a symbolic link.
* `updatedSince` + 

> Inclusive lower boundary for the file modification stamp expressed as either a DateTime instance or a String in ISO-8601 format.

Reviewer: ```suggestion
An inclusive lower boundary for the file modification stamp expressed as either a DateTime instance or a string in ISO-8601 format.
```

* `updatedUntil` + 

> Inclusive upper boundary for the file modification stamp expressed as either a DateTime instance or a String in ISO-8601 format.

Reviewer: ```suggestion
An inclusive upper boundary for the file modification stamp expressed as either a DateTime instance or a string in ISO-8601 format.
```

      ...
</flow>
----

> === Inner, Non Reusable Matcher

Reviewer: ```suggestion
=== Inner, Nonreusable Matcher
```

> This is an example of an inner, non reusable, matcher:

Reviewer: ```suggestion
This is an example of an inner, nonreusable, matcher:
```
